### DISCUSSION TITLE
Error while reading chars string from s7 200 smart

### DISCUSSION THREAD

**Discussion by Diptee24** [gijzelaerr/python-snap7#419] [Q&A] [answered]
hi sir,
i am reading 6-7 address at a time using below format to read string. but i am getting error : b' ISO : Bad PDU format'. each plc address contains 15 byte.
client.read_area(snap7.types.Areas.DB,1,3200,15).decode() 

also if i use snap7.util.get_string(buffer1,3172,15) it gives me an error.
please suggest some solution.
thank you.

**Comment by nikteliy** [comment]
Could you capture packets using [Wireshark](https://www.wireshark.org/) and post them here in `.pcapng` format? This will help figure out what's wrong with the PDU format

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. S7-200 Smart has limited compatibility with snap7.
